src/main/steps/existing-application/post.test.ts

```diff
@@ -67,32 +74,71 @@ describe('ExistingApplicationPostController', () => {
   test.each([
     [ApplicationType.JOINT_APPLICATION, '2022-08-22', undefined],
     [ApplicationType.SOLE_APPLICATION, '2022-08-22', '2022-09-15'],
-  ])('Should be redirected to the current page for %s', async (applicationType, dateSubmitted, dateAosSubmitted) => {
+  ])(
+    'When New case chosen but unlinking not allowed (%s)',
+    async (applicationType, dateSubmitted, dateAosSubmitted) => {
+      const body = {
+        existingOrNewApplication: existingOrNew.New,
+      };
+
+      const caseApiMockFnReturnValue = caseApiMockFn({ applicationType, dateSubmitted, dateAosSubmitted, id: '1234' });
```

Comment:
this variable name is a bit confusing, because returnValue gives you the impression that it's a value object, when we're using it more to mock caseApi functions. why not just call it something like mockCaseApi?